src/transforms/aws_ec2_metadata.rs

```diff
@@ -186,14 +186,24 @@ impl TaskTransform for Ec2MetadataTransform {
 
 impl Ec2MetadataTransform {
     fn transform_one(&mut self, mut event: Event) -> Event {
-        let log = event.as_mut_log();
-
         if let Some(read_ref) = self.state.read() {
-            read_ref.into_iter().for_each(|(k, v)| {
-                if let Some(value) = v.get_one() {
-                    log.insert(k.clone(), value.clone());
+            match event {
+                Event::Log(ref mut log) => {
+                    read_ref.into_iter().for_each(|(k, v)| {
+                        if let Some(value) = v.get_one() {
+                            log.insert(k.clone(), value.clone());
+                        }
+                    });
+                }
+                Event::Metric(ref mut metric) => {
+                    read_ref.into_iter().for_each(|(k, v)| {
+                        if let Some(value) = v.get_one() {
+                            metric
+                                .insert_tag(k.clone(), String::from_utf8(value.to_vec()).unwrap());
```

Comment:
I share this concern. Two thoughts, to add on. First, in general calling `unwrap` is something we want to avoid, since it kills the vector. There are some cases where we absolutely can't recover but here I think we could easily either do a lossy conversion like @bruceg is suggesting or just not in the logs that a value couldn't be decoded. Second, even if it's documented on the AWS side today as always being UTF-8 having an `unwrap` here is a footgun in case that ever changes or AWS bugs out.

FWIW my preference is for `String::from_utf8_lossy` here.

Reply:
Appreciate the explanations. I'll move forward with `from_utf8_lossy`.